fili-core/src/test/groovy/com/yahoo/bard/webservice/web/apirequest/generator/intervals/UtcBasedIntervalGeneratorSpec.groovy

```diff
@@ -0,0 +1,107 @@
+import org.joda.time.Interval
+import org.joda.time.Period
+import org.joda.time.format.DateTimeFormatter
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class UtcBasedIntervalGeneratorSpec extends Specification {
+
+    UtcBasedIntervalGenerator generator = new UtcBasedIntervalGenerator()
+    DateTimeFormatter formatter = DateTimeFormatterFactory.FULLY_OPTIONAL_DATETIME_FORMATTER
+
+    static DateTime now
+    static DateTimeZone original
+    static Period oneDay = new Period("P1D")
+
+    def setupSpec() {
+        original = DateTimeZone.getDefault()
+        DateTimeZone.setDefault(DateTimeZone.UTC)
+        now = new DateTime(0)
+    }
+
+    def cleanupSpec() {
+        DateTimeZone.setDefault(original)
+    }
+
+    def "Validate fails if the interval isn't set."() {
+        setup:
+        DataApiRequestBuilder builder = Mock(DataApiRequestBuilder)
+        Optional<Granularity> granularity = Optional.empty()
+        Granularity g = DefaultTimeGrain.DAY
+
+        when:
+        generator.validateGranularityPresent(builder)
+
+        then:
+        1 * builder.isGranularityInitialized() >> false
+        Exception e = thrown(UnsatisfiedApiRequestConstraintsException)
+
+        when:
+        generator.validateGranularityPresent(builder)
+
+        then:
+        1 * builder.isGranularityInitialized() >> true
+        1 * builder.getGranularityIfInitialized() >> Optional.empty()
+        e = thrown(BadApiRequestException)
+        e.getMessage().equals(UtcBasedIntervalGenerator.GRANULARITY_REQUIRED)
+
+        when:
+        Granularity actual = generator.validateGranularityPresent(builder)
+
+        then:
+        1 * builder.isGranularityInitialized() >> true
+        builder.getGranularityIfInitialized() >> Optional.of(g)
+        actual == g
+    }
+
+
+    def "Validate fails if the interval doesn't align"() {
+        setup:
+        Granularity g = DefaultTimeGrain.DAY
+        DateTime dateTime = new DateTime(-1)
+        Interval i = new Interval(dateTime, g.getPeriod())
+
+        when:
+        UtcBasedIntervalGenerator.validateTimeAlignment(g, Collections.singletonList(i))
+
+        then:
+        Exception e = thrown(BadApiRequestException)
+        e.getMessage().contains(i.toString())
+    }
+
+    @Unroll
+    def "GenerateIntervals for #request produces #intervals"() {
+        setup:
+        Granularity g = DefaultTimeGrain.DAY
+        List<Interval> result = UtcBasedIntervalGenerator.generateIntervals(now, request, g, formatter)
+
+        expect:
+        intervals == result
+
+        where:
+        request             | intervals
+        "P1D/Current"       | [new Interval(oneDay, now)]
+        "Current/Next"      | [new Interval(now, oneDay)]
+        "1970/P1D,1971/P1D" | [new Interval(now, oneDay), new Interval(now.plusYears(1), oneDay)]
+        "1970/1980"         | [new Interval(now, now.plusYears(10))]
+        "1970/RRULE=FREQ=MONTHLY;BYMONTHDAY=-1;COUNT=3"         |
```

Comment:
is this valid to have current/next macro with RRULE ?
something like current/RRULE=[FREQ=....]